macOSUpgrade.sh

```diff
@@ -362,6 +371,16 @@ if [[ ${pwrStatus} == "OK" ]] && [[ ${spaceStatus} == "OK" ]]; then
         /Library/Application\ Support/JAMF/bin/jamfHelper.app/Contents/MacOS/jamfHelper -windowType utility -title "$title" -icon "$icon" -heading "$heading" -description "$description" -iconSize 100 &
         jamfHelperPID=$!
     fi
+	##Promote Current User to admin if needed
+	currentUser=$(/usr/bin/python -c 'from SystemConfiguration import SCDynamicStoreCopyConsoleUser; import sys; username = (SCDynamicStoreCopyConsoleUser(None, None, None) or [None])[0]; username = [username,""][username in [u"loginwindow", None, u""]]; sys.stdout.write(username + "\n");')
```

Comment:
We can also get currentUser with the $3 parameter.
why do you use python script?
Please let us know if there is a reason.

Reply:
> As some people run this script without using Jamf

This project depends on Jamf. 
Still need we to think about this use case?

Reply:
No it does not. You can hard code variables as detailed in the script and you can change out the jamfhelper for something else like osascript or cocoadialogue

Reply:
Also in the past there have been times that $3 has been set incorrectly and as such scripts have failed. In something as important as an OS upgrade do you want to trust that $3 is set correctly?

Reply:
Basically this project is assumed to be executed as a script of jamf.

> you can change out the jamfhelper for something else like osascript or cocoadialogue

In these cases, I think that it is true that each counterpart turns `$3` into python script.

---

> Also in the past there have been times that `$3` has been set incorrectly and as such scripts have failed.

I think that this reason for this is a problem to be discussed fully.

> https://www.jamf.com/jamf-nation/articles/146/script-parameters
> Username, specifically:
> - If the script is run with a login or logout policy—Username of the account used to log in or out of the computer
> - If the script is run from Self Service—Username of the account used to logged in to Self Service

If you trust this explanation, when executing script of this project with checkin event, I thought that there is a possibility that the user name might not be set.
Therefore I thought that using `$3` was reasonable.